### PR TITLE
Feature/rw edit shipment date

### LINKED ISSUE #10: User Image
- update user schema (and all related files) with column for image
- write method for adding an image 
   - 'mutation'
   - changes DB for that user 
   - 'update' 
- test in GraphQL GUI 
   - [x] new user (null image col) add image 
   - [x] existing user with image update image 
- [ ] update seed data

### LINKED ISSUE #41: Create Coming Soon UI for Order History page
- [ ] Create new page for order history form 
- [x] On click of "Order History" button on profile page, route to this page 
[Crate (2).pdf](https://github.com/Strompy/Crate/files/5362314/Crate.2.pdf)

**Modified for this sprint: order history page will just be an "Under construction" or "Coming soon" page since we don't have time to implement functionality**

### LINKED ISSUE #43: Add ability to edit shipment date
- To Profile page, add "Edit ship date" button, which will be a date picker

When user submits that new date:
- Make call to back-end to update subscription info
- Should receive back subscription info & then dispatch that to update store 
- Add into mapStateToProps function to get subscription info from store so it can be rendered 

**Note: unsure of how this data will be structured/stored on back-end so this may need adjustments**

### PR DESCRIPTION
**Pull request by JoshSevy**
# Crate Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
- Add calendar UI
- Add Under Construction page for order history
- functionality for date selection for editing shipment date on subscriptions
- Last push bug fixes
- Date picker saving to server
- Lint code
- Fix bugs with profile edit image
- minor tweeks to ui

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other, Please explain: Clean up code

## Has This Been Tested?
- [ ] No
- [x] Yes

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
User tested/Browser Tested
# Message for reviewer:
Great Work everyone! You all killed it.
# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/web/src/modules/subscription/api/actions.js [hunk 1]
```diff
@@ -141,3 +142,34 @@ export function remove(variables) {
     }))
   }
 }
+
+
+//Change date for selected subscription
+export function setSubscriptionDate(id, date) {
+  return dispatch => {
```
Comment (rwilliams659) [inline code/web/src/modules/subscription/api/actions.js:149]: So glad we had time to get this hooked up to the BE

File: code/web/src/modules/subscription/api/actions.js [hunk 2]
```diff
@@ -29,7 +30,7 @@ export function getList(isLoading = true) {
 
     return axios.post(routeApi, query({
       operation: 'subscriptions',
-      fields: ['id', 'user { name, email }', 'crate { id, name, description }', 'createdAt']
+      fields: ['id', 'date', 'user { name, email }', 'crate { id, name, description }', 'createdAt']
```
Comment (Abdeboskey) [inline code/web/src/modules/subscription/api/actions.js:33]: Thanks again back end! Y'all made this super simple for us!

File: code/web/src/modules/user/EditProfile.js [hunk 3]
```diff
@@ -128,12 +126,14 @@ class EditProfile extends Component {
 
         window.setTimeout(() => {
           this.props.messageHide()
-          
         }, 5000)
       })
   }
 
   render() {
+    if (this.state.submitted) {
```
Comment (rwilliams659) [inline code/web/src/modules/user/EditProfile.js:134]: Awesome job fixing this up & getting the new image to display correctly

File: code/web/src/modules/user/Profile.js [hunk 4]
```diff
@@ -11,128 +11,173 @@ import { H3, H4, H5, H6 } from '../../ui/typography'
...
+      date: ''
+    }
+  }
+
+  onChange = (event) => {
+    const date = event.target.value
+    this.setState({ date })
+  }
+
+  formatDate(date) {
```
Comment (rwilliams659) [inline code/web/src/modules/user/Profile.js:40]: Love the simple date formatter!

**Review by rwilliams659 — APPROVED** [review]
Great job team! We're done!

File: code/web/src/modules/user/EditProfile.js [hunk 5]
```diff
@@ -128,12 +126,14 @@ class EditProfile extends Component {
...
+      return <Redirect to={userRoutes.profile.path} />
+    }
```
Comment (Abdeboskey) [inline code/web/src/modules/user/EditProfile.js:136]: Thanks for trusting me on this one!

**Review by Abdeboskey — APPROVED** [review]
So fun! What a rush. Excellent work, everybody.

File: code/web/src/modules/subscription/api/actions.js (hunk 2 above)
Comment (Strompy) [reply code/web/src/modules/subscription/api/actions.js:33]: So glad to hear that!

File: code/web/src/modules/subscription/api/actions.js [hunk 6]
```diff
@@ -141,3 +142,34 @@ export function remove(variables) {
...
+
+
+//Change date for selected subscription
+export function setSubscriptionDate(id, date) {
+  return dispatch => {
+
+    return axios.post(routeApi, mutation({
+      operation: 'subscriptionDateUpdate',
+      variables: { id, date },
+      fields: ['id', 'date']
```
Comment (Strompy) [inline code/web/src/modules/subscription/api/actions.js:154]: Perfect!

File: code/web/src/modules/user/orderHistory.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      </GridCell>
+    </Grid>
+
+    <Grid>
+      <GridCell style={{ textAlign: 'center' }}>
+        <p style={{ textAlign: 'center', marginTop: '2em', marginBottom: '2em' }}>
+          <img src={`${APP_URL}/images/under-construction.png`} alt="under construction" style={{ height: '20em' }} />
+        </p>
+
+        <H3 font="secondary" style={{ marginBottom: '1em' }}>This page is currently under construction. Please check back later!</H3>
```
Comment (Strompy) [inline code/web/src/modules/user/orderHistory.js:39]: Nice job getting the order history page started!

**Review by Strompy — APPROVED** [review]
Looks great!

**Pull request merged**